src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,114 @@
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void beforeEach() {
+        user = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+    }
+
+    @AfterEach
+    void afterEach() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_NullUser_false() {
+        assertThrows(InvalidUserDataException.class, () ->
+                service.register(null), " expected "
+                + InvalidUserDataException.class.getName()
+                + " to be thrown for less length login "
+                + user
+                + " but it wasn't");
+    }
+
+    @Test
+    void register_NullLogin_false() {
+        user.setLogin(null);
+        assertThrows(InvalidUserDataException.class, () ->
+                service.register(user)," expected "
+                + InvalidUserDataException.class.getName()
+                + " to be thrown for less length login "
+                + user.getLogin()
+                + " but it wasn't");
+    }
+
+    @Test
+    void register_NullAge_false() {
+        user.setAge(null);
+        assertThrows(InvalidUserDataException.class, () ->
+                service.register(user)," expected "
+                + InvalidUserDataException.class.getName()
+                + " to be thrown for null age "
+                + user.getAge()
+                + " but it wasn't");
+    }
+
+    @Test
+    void register_NullPass_false() {
+        user.setPassword(null);
+        assertThrows(InvalidUserDataException.class, () ->
+                service.register(user)," expected "
+                + InvalidUserDataException.class.getName()
+                + " to be thrown for null password "
+                + user.getPassword()
+                + " but it wasn't");
+    }
+
+    @Test
+    void register_loginDublicate_false() {
+        Storage.people.add(user);
+        User newUser = new User(user.getLogin(), VALID_PASSWORD, VALID_AGE);
+        assertThrows(InvalidUserDataException.class, () -> service.register(newUser),
+                "Expected" + InvalidUserDataException.class.getName()
+                + "to be thrown when registering a user with a duplicate login"
+                + newUser.getLogin()
+                + "but it was not thrown");
+    }
+
+    @Test
+    void register_passwordShort_false() {
+        user.setPassword(WRONG_PASSWORD);
+        assertThrows(InvalidUserDataException.class, () -> service.register(user),
+                " expected " + InvalidUserDataException.class.getName()
+                    + " to be thrown for age " + user.getPassword() + " but it wasn't");
+    }
+
+    @Test
+    void register_userAgeUnder18_false() {
+        user.setAge(LOWER_18);
+        assertThrows(InvalidUserDataException.class, () -> service.register(user),
+                " expected " + InvalidUserDataException.class.getName()
+                    + " to be thrown for less age "
+                    + user.getAge()
```

Comment:
something like this would be a bit better, same for others
```suggestion
                    + " to be thrown for age = " + user.getAge()
```